### DISCUSSION TITLE
Callback For A Skipped Event

### DISCUSSION THREAD

**Discussion by infohash** [apache/airflow#37164] [Ideas] [answered]
There are five types of task events that can trigger a callback:

Name | Description
-- | --
on_success_callback | Invoked when the task succeeds
on_failure_callback | Invoked when the task fails
sla_miss_callback | Invoked when a task misses its defined SLA
on_retry_callback | Invoked when the task is up for retry
on_execute_callback | Invoked right before the task begins executing.

It would be good to have `on_skip_callback` which should be invoked when the task is skipped. `S3KeySensor` can be skipped if it does not find the key in S3 by setting `soft_fail` to `True`. In this case, `on_skip_callback` can notify the user about its status.

**Comment by Taragolis** [comment, marked as answer]
[`on_skipped_callback`](https://github.com/apache/airflow/pull/36374) will be a part of the Airflow 2.9
